tools/pipelines/templates/include-log-test-failures.yml

```diff
@@ -5,13 +5,32 @@ parameters:
 - name: buildDirectory
   type: string
 
+# A list of directories (under the buildDirectory) to run the PublishTestResults task on in separate steps.
+# Used to avoid the force merge limit of 100 result files.
+- name: testResultDirs
+  type: object
+  default:
+  - nyc
+
 # Log test failures
 steps:
+  # Test - Upload results
+  - ${{ each testResultDir in parameters.testResultDirs }}:
+    - task: PublishTestResults@2
+      displayName: Publish Test Results in ${{ testResultDir }}
+      inputs:
+        testResultsFormat: 'JUnit'
+        testResultsFiles: '**/*junit-report.xml'
+        searchFolder: ${{ parameters.buildDirectory }}/${{ testResultDir }}
+        mergeTestResults: false
+      condition: and(succeededOrFailed(), eq(variables['startTest'], 'true'))
```

Comment:
I don't remember if accessing variables defined outside the template works. For robustness, make it a parameter? Or maybe easier, wrap the "call" to the template with an `if` that checks for `startTest` so the steps here don't need to care.

Reply:
I have tested it and it can access variables.  Because of binding, values that is determined at run time can't be a parameters or if.